File: yurt-system/yurt-system-server/src/main/java/com/github/liulus/yurt/system/service/MenuService.java
package com.github.liulus.yurt.system.service;

import com.github.liulus.yurt.system.model.dto.MenuDTO;
import com.github.liulus.yurt.system.model.entity.Menu;

import java.util.List;

/**
 * User : liulu
 * Date : 2017/7/10 10:50
 * version $Id: MenuService.java, v 0.1 Exp $
 */
public interface MenuService {

    /**
     * 根据Id查询菜单
     *
     * @param id 主键
     * @return 菜单
     */
    Menu findById(Long id);

    /**
     * 构建菜单树
     *
     * @param filterDisabled 是否过滤禁用菜单
     * @param filterEmpty    是否过滤子菜单为空并且没有url
     * @return Detail
     */
    List<MenuDTO.Detail> buildMenuTree(boolean filterDisabled, boolean filterEmpty);

    /**
     * 增加菜单
     *
     * @param add Menu
     */
    Long insert(MenuDTO.Add add);

    /**
     * 更新菜单
     *
     * @param update Menu
     */
    int update(MenuDTO.Update update);

    /**
     * 删除菜单
     *
     * @param id id
     */
    int delete(Long id);


    /**
     * 改变菜单状态, 启用->禁用, 禁用->启用
     *
     * @param id id
     */
    void changeStatus(Long id);

}

File: yurt-system/yurt-system-server/src/main/kotlin/com/github/liulus/yurt/system/service/MenuService.kt
package com.github.liulus.yurt.system.service

import com.github.liulus.yurt.system.model.dto.MenuDTO
import com.github.liulus.yurt.system.model.entity.Menu

/**
 * User : liulu
 * Date : 2017/7/10 10:50
 * version $Id: MenuService.java, v 0.1 Exp $
 */
interface MenuService {
    /**
     * 根据Id查询菜单
     *
     * @param id 主键
     * @return 菜单
     */
    fun findById(id: Long?): Menu?

    /**
     * 查询所以的菜单,包含资源 并构建为树状结构
     * @return Detail
     */
    fun buildAllMenuTree(): List<MenuDTO.Detail>

    /**
     * 查询用户的菜单
     * @return Detail
     */
    fun findUserMenus(userId: Long): List<MenuDTO.Detail>

    /**
     * 增加菜单
     * @param add Menu
     */
    fun insert(add: MenuDTO.Add): Long

    /**
     * 更新菜单
     * @param update Menu
     */
    fun update(update: MenuDTO.Update): Int

    /**
     * 删除菜单
     * @param id id
     */
    fun delete(id: Long): Int

    /**
     * 改变菜单状态, 启用->禁用, 禁用->启用
     * @param id id
     */
    fun changeStatus(id: Long)
}